File: src/2500.delete-greatest-value-in-each-row.2585/solution.ts
function deleteGreatestValue (grid: number[][]): number {
    for (const row of grid) {
        row.sort((a, b) => a - b);
    }
    let result = 0;
    const M = grid.length;
    const N = grid[0].length;
    for (let j = 0; j < N; j++) {
        let max = 0;
        for (let i = 0; i < M; i++) {
            max = Math.max(max, grid[i][j]);
        }
        result += max;
    }

    return result;
}

File: src/2500.delete-greatest-value-in-each-row.2585/solution.java
class Solution {
    public int deleteGreatestValue(int[][] grid) {
        int M = grid.length;
        int N = grid[0].length;
        for (int i = 0; i < grid.length; i++) {
            Arrays.sort(grid[i]);
        }
        int result = 0;
        for (int j = 0; j < N; j++) {
            int max = 0;
            for (int[] row : grid) {
                max = Math.max(max, row[j]);
            }
            result += max;
        }
        return result;
    }
}